### ISSUE TITLE
Incompatible CMake compile options for MSVC

### ISSUE THREAD

**Issue by paamand** [pybind/pybind11#1532]
pybind11Tools.cmake line 214 and 220 adds the "/MP" compile option. However, this option is incompatible with the default compile options /Gm (minimal rebuild) for MSVC. Result is compiler warning.

Solution: Only add the /MP option to DEBUG build.
Replace line 212 through 221 with
```
target_compile_options(${target_name} PRIVATE /bigobj)
if(CMAKE_VERSION VERSION_LESS 3.11)
  target_compile_options(${target_name} PRIVATE $<$<CONFIG:RELEASE>:/MP>)
else()
  # Only set these options for C++ files.  This is important so that, for
  # instance, projects that include other types of source files like CUDA
  # .cu files don't get these options propagated to nvcc since that would
  # cause the build to fail.
  target_compile_options(${target_name} PRIVATE $<$<CONFIG:RELEASE>:$<$<COMPILE_LANGUAGE:CXX>:/MP>>)
endif()
```

**Comment by paamand** [comment]
Fixed in pull request #1533 
https://github.com/pybind/pybind11/pull/1533

**Comment by bstaletic** [comment]
The pull request has been merged. Closing.
